core/unix/os.c

```diff
@@ -7416,6 +7422,52 @@ pre_system_call(dcontext_t *dcontext)
                                      dcontext->sys_num);
         break;
     }
+#    ifdef LINUX
+    case SYS_ppoll: {
+        kernel_sigset_t *mask = (kernel_sigset_t *)sys_param(dcontext, 3);
+        size_t sizemask = (size_t)sys_param(dcontext, 4);
+        kernel_sigset_t nullsigmask;
+        memset(&nullsigmask, 0, sizeof(nullsigmask));
+        /* original app's sigmask parameter is now NULL effectively making the syscall
+         * a non p* version, and the mask's semantics are emulated by DR instead. */
+        set_syscall_param(dcontext, 3, *(reg_t *)&nullsigmask);
+        if (handle_pre_extended_syscall_sigmasks(dcontext, mask, sizemask)) {
+            /* in old kernels with sizeof(kernel_sigset_t) != sizemask, we're forcing
+             * failure. we're already violating app transparency in other places in DR */
+            set_failure_return_val(dcontext, EINTR);
+        }
+        break;
+    }
+    case SYS_pselect6: {
+        typedef const struct {
+            kernel_sigset_t *sigmask;
+            size_t sizemask;
+        } data_t;
+        data_t *data = (data_t *)sys_param(dcontext, 5);
+        kernel_sigset_t *mask = (kernel_sigset_t *)data->sigmask;
```

Comment:
Be safe: safe_read instead of directly reading these two fields

Reply:
Ok